src/GuiBase/KeyFrameEditor/KeyFrameEditorFrame.cpp

```diff
@@ -0,0 +1,1220 @@
+        }
+
+        if ( m_ui != nullptr )
+        {
+            display( m_ui->m_posX, true );
+            display( m_ui->m_posY, true );
+            display( m_ui->m_posZ, true );
+            display( m_ui->m_posW, true );
+            m_ui->m_posX->setText( "x" );
+            m_ui->m_posY->setText( "y" );
+            m_ui->m_posZ->setText( "z" );
+            m_ui->m_posW->setText( "w" );
+        }
+    }
+    else if ( auto kf = dynamic_cast<KeyFramedValue<Ra::Core::Utils::Color>*>( m_value ) )
+    {
+        m_curveControlPoints[0].reserve( times.size() );
+        m_curveControlPoints[1].reserve( times.size() );
+        m_curveControlPoints[2].reserve( times.size() );
+        m_curveControlPoints[3].reserve( times.size() );
+        for ( auto t : times )
+        {
+            const auto f = kf->at( t );
+            m_curveControlPoints[0].push_back( {t, f( 0 )} );
+            m_curveControlPoints[1].push_back( {t, f( 1 )} );
+            m_curveControlPoints[2].push_back( {t, f( 2 )} );
+            m_curveControlPoints[3].push_back( {t, f( 3 )} );
+        }
+
+        if ( m_ui != nullptr )
+        {
+            display( m_ui->m_posX, true );
+            display( m_ui->m_posY, true );
+            display( m_ui->m_posZ, true );
+            display( m_ui->m_posW, true );
+            m_ui->m_posX->setText( "r" );
+            m_ui->m_posY->setText( "g" );
+            m_ui->m_posZ->setText( "b" );
+            m_ui->m_posW->setText( "a" );
+        }
+    }
+    else if ( auto kf = dynamic_cast<KeyFramedValue<Ra::Core::Quaternion>*>( m_value ) )
+    {
+        m_curveControlPoints[4].reserve( times.size() );
+        m_curveControlPoints[5].reserve( times.size() );
+        m_curveControlPoints[6].reserve( times.size() );
+        for ( auto t : times )
+        {
+            const auto f = kf->at( t ).matrix().eulerAngles( 0, 1, 2 );
+            m_curveControlPoints[4].push_back( {t, f.x()} );
+            m_curveControlPoints[5].push_back( {t, f.y()} );
+            m_curveControlPoints[6].push_back( {t, f.z()} );
+        }
+
+        if ( m_ui != nullptr )
+        {
+            display( m_ui->m_rotX, true );
+            display( m_ui->m_rotY, true );
+            display( m_ui->m_rotZ, true );
+        }
+    }
+    else if ( auto kf = dynamic_cast<KeyFramedValue<Ra::Core::Transform>*>( m_value ) )
+    {
+        m_curveControlPoints[0].reserve( times.size() );
+        m_curveControlPoints[1].reserve( times.size() );
+        m_curveControlPoints[2].reserve( times.size() );
+        m_curveControlPoints[4].reserve( times.size() );
+        m_curveControlPoints[5].reserve( times.size() );
+        m_curveControlPoints[6].reserve( times.size() );
+        m_curveControlPoints[7].reserve( times.size() );
+        m_curveControlPoints[8].reserve( times.size() );
+        m_curveControlPoints[9].reserve( times.size() );
+        for ( auto t : times )
```

Comment:
well, there are a lot of manual (almost)loop unroll like this in this code ... This does not facilitate the readings.
If some others are less trivial, put these king of repetition int a loop with the minimum number of lines. It will facilitate code maintenance.

Reply:
Well, since each type has its mapping, I preffered letting this one unrolled so that it is clear.